<?xml version="5.0"?>
<launch>

<arg name="rate" default="1.0" />
<arg name="gui" default="false" />
<arg name="record" default="false" />
<arg name="output_name" default="radarays_output" />
<arg name="print_similarity" default="false" />


<arg name="meshfile" default="/home/amock/datasets/kaist02_mesh_v6M_f11M.ply" />
<!-- <arg name="meshfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc03_mesh_lvr2.ply" /> -->

<arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist01_micp_lvr2.bag" />
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist02_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/KAIST/mulran_kaist03_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc01_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc02_micp_lvr2.bag" /> -->
<!-- <arg name="bagfile" default="/home/amock/datasets/micp_mulran/DCC/mulran_dcc03_micp_lvr2.bag" /> -->

<arg name="radarparams" default="$(find radarays_ros)/cfg/mulran_kaist_flex_dyncfg.yaml" />


<node pkg="rosbag" type="play" name="rosbag_play" args="$(arg bagfile) -r $(arg rate) --pause" />



<node pkg="radarays_ros" type="radar_simulator_flex" name="radar_simulator_flex" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <param name="sync_topic" type="string" value="/Navtech/Polar" />
    <param name="sensor_frame" type="string" value="radar_polar" />
    # Load material file here
    <rosparam command="load" file="$(find radarays_ros)/config/mulran_kaist02.yaml" />
</node>

<node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" 
    args="load /radar_simulator_flex $(arg radarparams)" />


<node pkg="radar_tools" type="radar_img_to_pcl" name="radar_img_to_pcl_radarays" output="screen">
    <param name="frame" type="string" value="radar_polar" />
    <remap from="radar_image" to="/radar/image" />
    <rosparam command="load" param="radar" file="$(find radarays_ros)/config/navtech_mulran.yaml" />
</node>

<node pkg="radar_tools" type="radar_img_to_pcl" name="radar_img_to_pcl_gt" output="screen">
    <param name="frame" type="string" value="radar_polar" />
    <remap from="radar_image" to="/radar/image_gt" />
    <rosparam command="load" param="radar" file="$(find radarays_ros)/config/navtech_mulran.yaml" />
</node>

<group if="$(arg print_similarity)">
    <node pkg="radar_tools" type="compare_radar_images.py" name="compare_radar_images" output="screen">
        <param name="topic_in_1" value="/radar/image_gt" />
        <param name="topic_in_2" value="/radar/image" />
        <param name="topic_out" value="/real_to_sim_gap"/>
        <param name="time_sync" value="virtual" />
    </node>
</group>

<group if="$(arg record)">
    <node pkg="rosbag" type="record" name="rosbag_record" args="--regex '/radar/image(.*)' -O $(arg output_name)" />
</group>

<group if="$(arg gui)">
    <node pkg="rviz" type="rviz" name="rviz_navtech" args="-d $(find radarays_ros)/rviz/mulran_radarays.rviz">
        <rosparam param="rviz_map_plugin" subst_value="True">
            Map3D: $(arg meshfile)
        </rosparam>
    </node>

    <node pkg="rviz" type="rviz" name="rviz_radarays" args="-d $(find radarays_ros)/rviz/mulran_navtech.rviz">
        <rosparam param="rviz_map_plugin" subst_value="True">
            Map3D: $(arg meshfile)
        </rosparam>
    </node>
</group>

</launch>
